### PR TITLE
GDPR-70 Performing final check on deletion grant before deleting

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: src/main/java/net/syscon/prison/repository/OffenderDeletionRepository.java [hunk 1]
```diff
@@ -4,5 +4,5 @@
 
 public interface OffenderDeletionRepository {
 
-    Set<String> deleteOffender(String offenderNo);
+    Set<Long> deleteOffender(String offenderNo);
```
Comment (brightonsbox) [inline src/main/java/net/syscon/prison/repository/OffenderDeletionRepository.java:7]: Unrelated change correcting types of ids

File: src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/service/OffenderDeletionService.java [hunk 2]
```diff
@@ -1,35 +1,86 @@
...
+    private final OffenderAliasPendingDeletionRepository offenderAliasPendingDeletionRepository;
     private final OffenderDeletionRepository offenderDeletionRepository;
     private final DataComplianceEventPusher dataComplianceEventPusher;
     private final TelemetryClient telemetryClient;
 
-    public void deleteOffender(final String offenderNumber, final Long referralId) {
+    public void deleteOffender(final OffenderDeletionGrant grant) {
 
-        // TODO GDPR-70 Conduct final double-check that the record hasn't been updated before deleting
+        checkRequestedDeletion(grant);
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/service/OffenderDeletionService.java:41]: A sort of "Are you sure?" :-)

Comment (brightonsbox) [reply src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/service/OffenderDeletionService.java:41]: Yep 👍
